vector_double.cpp

```diff
@@ -0,0 +1,207 @@
+#include "vector_double.hpp"
+
+
+template <typename T>
+Vector<T>::Vector()
+{
+    size_v = 0;
+    space  = 0;
+    elem   = nullptr;
```

Comment:
we shouldn't initialize the vector to be empty, we should create a vector with some default size (16 maybe?)

Reply:
We should use member initialization lists for these stuff to make them shorter.
```cpp
template<typename T>
Vector<T>::Vector() : size_v(0), space(0), elem(nullptr) {}
```

Reply:
What about enum? Will that work or member initialization lists more easier?

Reply:
I don't see where an enum would fit into this -- can you please elaborate?

Reply:
Never mind the enum